### PR TITLE
Add allow_simultaneous_ips param for targets

### PR DESCRIPTION
**Pull request by timopollmeier**
**What**:
The create_target and modify_target commands now have an optional
parameter to set whether scanning multiple IPs per host simultaneously
is allowed.
The tests for the commands are updated accordingly.

**Why**:
To make the new option usable in python-gvm.

**How**:
Tested by running a program that uses the two commands, including the
new option.

**Checklist**:

- [x] Tests
- [x] [CHANGELOG](https://github.com/greenbone/python-gvm/blob/master/CHANGELOG.md) Entry
- [x] Documentation

### CODE REVIEW

**Review by bjoernricks — COMMENTED** [review]
I am not 100% sure about the variable name `allow_simult_ips_same_host`. But `allow_simultaneous_ips_same_host` gets a bit long. @y0urself can decide about that.

**Comment by y0urself** [comment]
> I am not 100% sure about the variable name `allow_simult_ips_same_host`. But `allow_simultaneous_ips_same_host` gets a bit long. @y0urself can decide about that.

Yeah would agree.
I would do something like `multiple_host_ips`. Any curiosities can be solved by the comment and documentation ...

**Comment by bjoernricks** [comment]
What about `allow_multiple_ips_per_host`?

**Comment by y0urself** [comment]
I don't think that `allow` needs to be mentioned here ... as its a boolean that can be `True` or `False` it might be enough to make it `multiple_ips_per_host`. For me allow refers more to Yes or No, not False or True.

**Comment by bjoernricks** [comment]
You are right. I am fine with `multiple_ips_per_host`. @timopollmeier do you agree?

File: gvm/protocols/gmpv214/gmpv214.py [hunk 1]
```diff
@@ -226,6 +226,132 @@ def create_override(
...
+            hosts: List of hosts addresses to scan
+            exclude_hosts: List of hosts addresses to exclude from scan
+            comment: Comment for the target
+            ssh_credential_id: UUID of a ssh credential to use on target
+            ssh_credential_port: The port to use for ssh credential
+            smb_credential_id: UUID of a smb credential to use on target
+            snmp_credential_id: UUID of a snmp credential to use on target
+            esxi_credential_id: UUID of a esxi credential to use on target
+            alive_test: Which alive test to use
+            allow_simult_ips_same_host: Whether to scan multiple IPs of the
```
Comment (y0urself) [inline gvm/protocols/gmpv214/gmpv214.py:265]:
```suggestion
            multiple_ips_per_host: Whether to scan multiple IPs of the
```

File: gvm/protocols/gmpv214/gmpv214.py [hunk 2]
```diff
@@ -226,6 +226,132 @@ def create_override(
...
+        hosts: Optional[List[str]] = None,
+        comment: Optional[str] = None,
+        exclude_hosts: Optional[List[str]] = None,
+        ssh_credential_id: Optional[str] = None,
+        ssh_credential_port: Optional[int] = None,
+        smb_credential_id: Optional[str] = None,
+        esxi_credential_id: Optional[str] = None,
+        snmp_credential_id: Optional[str] = None,
+        alive_test: Optional[AliveTest] = None,
+        allow_simult_ips_same_host: Optional[bool] = None,
```
Comment (y0urself) [inline gvm/protocols/gmpv214/gmpv214.py:244]:
```suggestion
        multiple_ips_per_host: Optional[bool] = None,
```

File: gvm/protocols/gmpv214/gmpv214.py [hunk 3]
```diff
@@ -226,6 +226,132 @@ def create_override(
...
+            if not isinstance(alive_test, AliveTest):
+                raise InvalidArgumentType(
+                    function=self.create_target.__name__,
+                    argument='alive_test',
+                    arg_type=AliveTest.__name__,
+                )
+
+            cmd.add_element("alive_tests", alive_test.value)
+
+        if allow_simult_ips_same_host is not None:
```
Comment (y0urself) [inline gvm/protocols/gmpv214/gmpv214.py:331]:
```suggestion
        if multiple_ips_per_host is not None:
```

File: gvm/protocols/gmpv214/gmpv214.py [hunk 4]
```diff
@@ -226,6 +226,132 @@ def create_override(
...
+            cmd.add_element(
+                "allow_simult_ips_same_host",
+                _to_bool(allow_simult_ips_same_host),
```
Comment (y0urself) [inline gvm/protocols/gmpv214/gmpv214.py:334]:
```suggestion
                _to_bool(multiple_ips_per_host),
```

File: gvm/protocols/gmpv214/gmpv214.py [hunk 5]
```diff
@@ -389,6 +515,118 @@ def modify_override(
...
+        comment: Optional[str] = None,
+        hosts: Optional[List[str]] = None,
+        exclude_hosts: Optional[List[str]] = None,
+        ssh_credential_id: Optional[str] = None,
+        ssh_credential_port: Optional[bool] = None,
+        smb_credential_id: Optional[str] = None,
+        esxi_credential_id: Optional[str] = None,
+        snmp_credential_id: Optional[str] = None,
+        alive_test: Optional[AliveTest] = None,
+        allow_simult_ips_same_host: Optional[bool] = None,
```
Comment (y0urself) [inline gvm/protocols/gmpv214/gmpv214.py:532]:
```suggestion
        multiple_ips_per_host: Optional[bool] = None,
```

File: gvm/protocols/gmpv214/gmpv214.py [hunk 6]
```diff
@@ -389,6 +515,118 @@ def modify_override(
...
+            hosts: List of target hosts.
+            exclude_hosts: A list of hosts to exclude.
+            ssh_credential_id: UUID of SSH credential to use on target.
+            ssh_credential_port: The port to use for ssh credential
+            smb_credential_id: UUID of SMB credential to use on target.
+            esxi_credential_id: UUID of ESXi credential to use on target.
+            snmp_credential_id: UUID of SNMP credential to use on target.
+            port_list_id: UUID of port list describing ports to scan.
+            alive_test: Which alive tests to use.
+            allow_simult_ips_same_host: Whether to scan multiple IPs of the
```
Comment (y0urself) [inline gvm/protocols/gmpv214/gmpv214.py:552]:
```suggestion
            multiple_ips_per_host: Whether to scan multiple IPs of the
```

File: gvm/protocols/gmpv214/gmpv214.py [hunk 7]
```diff
@@ -389,6 +515,118 @@ def modify_override(
...
+        if smb_credential_id:
+            cmd.add_element("smb_credential", attrs={"id": smb_credential_id})
+
+        if esxi_credential_id:
+            cmd.add_element("esxi_credential", attrs={"id": esxi_credential_id})
+
+        if snmp_credential_id:
+            cmd.add_element("snmp_credential", attrs={"id": snmp_credential_id})
+
+        if allow_simult_ips_same_host is not None:
```
Comment (y0urself) [inline gvm/protocols/gmpv214/gmpv214.py:609]:
```suggestion
        if multiple_ips_per_host is not None:
```

File: gvm/protocols/gmpv214/gmpv214.py [hunk 8]
```diff
@@ -389,6 +515,118 @@ def modify_override(
...
+            cmd.add_element(
+                "allow_simult_ips_same_host",
+                _to_bool(allow_simult_ips_same_host),
```
Comment (y0urself) [inline gvm/protocols/gmpv214/gmpv214.py:612]:
```suggestion
                _to_bool(multiple_ips_per_host),
```

File: tests/protocols/gmpv214/testcmds/test_create_target.py [hunk 9]
```diff
@@ -0,0 +1,268 @@
...
+            '<hosts>foo</hosts>'
+            '<alive_tests>ICMP Ping</alive_tests>'
+            '</create_target>'
+        )
+
+    def test_create_target_invalid_alive_tests(self):
+        with self.assertRaises(InvalidArgumentType):
+            self.gmp.create_target('foo', hosts=['foo'], alive_test='foo')
+
+    def test_create_target_with_allow_simult_ips_same_host(self):
```
Comment (y0urself) [inline tests/protocols/gmpv214/testcmds/test_create_target.py:173]:
```suggestion
    def test_create_target_with_multiple_ips_per_host(self):
```

File: tests/protocols/gmpv214/testcmds/test_create_target.py [hunk 10]
```diff
@@ -0,0 +1,268 @@
...
+        self.gmp.create_target(
+            'foo',
+            hosts=['foo'],
+            allow_simult_ips_same_host=True
```
Comment (y0urself) [inline tests/protocols/gmpv214/testcmds/test_create_target.py:177]:
```suggestion
            multiple_ips_per_host=True
```

File: tests/protocols/gmpv214/testcmds/test_create_target.py [hunk 11]
```diff
@@ -0,0 +1,268 @@
...
+            '<name>foo</name>'
+            '<hosts>foo</hosts>'
+            '<allow_simult_ips_same_host>1</allow_simult_ips_same_host>'
+            '</create_target>'
+        )
+
+        self.gmp.create_target(
+            'foo',
+            hosts=['foo'],
+            allow_simult_ips_same_host=False
```
Comment (y0urself) [inline tests/protocols/gmpv214/testcmds/test_create_target.py:191]:
```suggestion
            multiple_ips_per_host=False
```

File: tests/protocols/gmpv214/testcmds/test_modify_target.py [hunk 12]
```diff
@@ -0,0 +1,219 @@
...
+            '<modify_target target_id="t1">'
+            '<alive_tests>ICMP Ping</alive_tests>'
+            '</modify_target>'
+        )
+
+    def test_modify_target_invalid_alive_tests(self):
+        with self.assertRaises(InvalidArgumentType):
+            self.gmp.modify_target(target_id='t1', alive_test='foo')
+
+    def test_modify_target_with_allow_simult_ips_same_host(self):
```
Comment (y0urself) [inline tests/protocols/gmpv214/testcmds/test_modify_target.py:157]:
```suggestion
    def test_modify_target_with_multiple_ips_per_host(self):
```

File: tests/protocols/gmpv214/testcmds/test_modify_target.py [hunk 13]
```diff
@@ -0,0 +1,219 @@
...
+        self.gmp.modify_target(target_id='t1', allow_simult_ips_same_host=True)
+
+        self.connection.send.has_been_called_with(
+            '<modify_target target_id="t1">'
+            '<allow_simult_ips_same_host>1</allow_simult_ips_same_host>'
+            '</modify_target>'
+        )
+
+        self.gmp.modify_target(target_id='t1', allow_simult_ips_same_host=False)
```
Comment (y0urself) [inline tests/protocols/gmpv214/testcmds/test_modify_target.py:166]:
```suggestion
        self.gmp.modify_target(target_id='t1', multiple_ips_per_host=False)
```

File: tests/protocols/gmpv214/testcmds/test_modify_target.py [hunk 14]
```diff
@@ -0,0 +1,219 @@
...
+            '<alive_tests>ICMP Ping</alive_tests>'
+            '</modify_target>'
+        )
+
+    def test_modify_target_invalid_alive_tests(self):
+        with self.assertRaises(InvalidArgumentType):
+            self.gmp.modify_target(target_id='t1', alive_test='foo')
+
+    def test_modify_target_with_allow_simult_ips_same_host(self):
+        self.gmp.modify_target(target_id='t1', allow_simult_ips_same_host=True)
```
Comment (y0urself) [inline tests/protocols/gmpv214/testcmds/test_modify_target.py:158]:
```suggestion
        self.gmp.modify_target(target_id='t1', multiple_ips_per_host=True)
```

**Review by y0urself — CHANGES_REQUESTED** [review]

File: tests/protocols/gmpv214/testcmds/__init__.py [hunk 15]
```diff
@@ -1,5 +1,5 @@
 # -*- coding utf-8 -*-
-# Copyright (C) 2020 Greenbone Networks GmbH
+# Copyright (C) 2021 Greenbone Networks GmbH
```
Comment (y0urself) [inline tests/protocols/gmpv214/testcmds/__init__.py:2]:
```suggestion
# Copyright (C) 2020-2021 Greenbone Networks GmbH
```

**Review by y0urself — CHANGES_REQUESTED** [review]

**Review by y0urself — APPROVED** [review]

**Pull request merged**